### PR TITLE
chore(ckbtc): property tests for event serialization and deserialization

### PR DESCRIPTION
**Pull request by lpahlavi**
[(XC-238)](https://dfinity.atlassian.net/browse/XC-238) Add property tests for the ckBTC event serialization/deserialization logic where events (both legacy and new) are arbitrarily generated from all possible event types. See [this comment](https://github.com/dfinity/ic/pull/3157#discussion_r1891578520).

### CODE REVIEW

File: rs/bitcoin/ckbtc/minter/src/test_fixtures.rs [hunk 1]
```diff
@@ -142,3 +142,309 @@ pub mod mock {
...
+            })
+        };
+    }
+
+    fn amount() -> impl Strategy<Value = Satoshi> {
+        1..10_000_000_000u64
+    }
+
+    fn txid() -> impl Strategy<Value = Txid> {
+        pvec(any::<u8>(), 32).prop_map(|bytes| {
```
Comment (gregorydemay) [inline rs/bitcoin/ckbtc/minter/src/test_fixtures.rs:189]: nit: With`uniform32(any::<u8>())` you directly obtain an array of the right size.

File: rs/bitcoin/ckbtc/minter/src/test_fixtures.rs [hunk 2]
```diff
@@ -142,3 +142,309 @@ pub mod mock {
...
+        option,
+        prelude::{any, Just, Strategy},
+        prop_oneof,
+    };
+    use serde_bytes::ByteBuf;
+
+    // Macro to simplify writing strategies that generate structs.
+    macro_rules! prop_struct {
+        ($struct_path:path { $($field_name:ident: $strategy:expr),* $(,)? }) => {
+            #[allow(unused_parens)]
```
Comment (gregorydemay) [inline rs/bitcoin/ckbtc/minter/src/test_fixtures.rs:175]: why is this needed? Is there any way we can remove `unused_parens`?

**Review by gregorydemay — APPROVED** [review]
Just had the time for a brief review, but LGTM!

File: rs/bitcoin/ckbtc/minter/src/test_fixtures.rs (hunk 2 above)
Comment (lpahlavi) [reply rs/bitcoin/ckbtc/minter/src/test_fixtures.rs:175]: Good question. It seems that this is a bug in the Rust compiler, see e.g. [this discussion](https://stackoverflow.com/questions/74983561/unnecessary-parens-warning-in-macro-call-that-needs-the-parens-is-this-a-bad) and the [linked GitHub issue](https://github.com/rust-lang/rust/issues/73068).

File: rs/bitcoin/ckbtc/minter/src/test_fixtures.rs (hunk 1 above)
Comment (lpahlavi) [reply rs/bitcoin/ckbtc/minter/src/test_fixtures.rs]: Good point, thanks for the suggestion!

**Review by ninegua — APPROVED** [review]
LGTM! Thanks
